pkg/core/resources/apis/mesh/proxytemplate_validator.go

```diff
@@ -29,10 +34,142 @@ func (t *ProxyTemplateResource) Validate() error {
 	return verr.OrNil()
 }
 
-func validateConfig(conf *v1alpha1.ProxyTemplate_Conf) validators.ValidationError {
+func validateConfig(conf *mesh_proto.ProxyTemplate_Conf) validators.ValidationError {
 	var verr validators.ValidationError
 	verr.Add(validateImports(conf.GetImports()))
 	verr.Add(validateResources(conf.GetResources()))
+	for i, modification := range conf.GetModifications() {
+		verr.AddErrorAt(validators.RootedAt("modifications").Index(i), validateModification(modification))
+	}
+	return verr
+}
+
+func validateModification(modification *mesh_proto.ProxyTemplate_Modifications) validators.ValidationError {
+	verr := validators.ValidationError{}
+	switch modification.Type.(type) {
+	case *mesh_proto.ProxyTemplate_Modifications_Cluster_:
+		verr.AddError("cluster", validateClusterModification(modification.GetCluster()))
+	case *mesh_proto.ProxyTemplate_Modifications_Listener_:
+		verr.AddError("listener", validateListenerModification(modification.GetListener()))
+	case *mesh_proto.ProxyTemplate_Modifications_NetworkFilter_:
+		verr.AddError("networkFilter", validateNetworkFilterModification(modification.GetNetworkFilter()))
+	case *mesh_proto.ProxyTemplate_Modifications_HttpFilter_:
+		verr.AddError("httpFilter", validateHTTPFilterModification(modification.GetHttpFilter()))
+	}
+	return verr
+}
+
+func validateHTTPFilterModification(filterMod *mesh_proto.ProxyTemplate_Modifications_HttpFilter) validators.ValidationError {
+	verr := validators.ValidationError{}
+	validateResource := func() {
+		if err := ValidateResourceYAML(&envoy_hcm.HttpFilter{}, filterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	}
+	switch filterMod.Operation {
+	case mesh_proto.OpAddFirst:
+		validateResource()
+	case mesh_proto.OpAddLast:
+		validateResource()
+	case mesh_proto.OpAddBefore:
+		if filterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty. You need to pick a filter before which this one will be added")
+		}
+		validateResource()
+	case mesh_proto.OpAddAfter:
+		if filterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty. You need to pick a filter after which this one will be added")
+		}
+		validateResource()
+	case mesh_proto.OpPatch:
+		if filterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty")
+		}
+		if err := ValidateResourceYAMLPatch(&envoy_hcm.HttpFilter{}, filterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	case mesh_proto.OpRemove:
+	default:
+		verr.AddViolation("operation", fmt.Sprintf("invalid operation. Available operations: %q, %q, %q, %q, %q", mesh_proto.OpAddFirst, mesh_proto.OpAddLast, mesh_proto.OpAddBefore, mesh_proto.OpAddAfter, mesh_proto.OpPatch))
```

Comment:
You missed `mesh_proto.OpRemove` here

Reply:
It's not missed, there is nothing to validate there :)